test/config-specs.js

```diff
@@ -0,0 +1,282 @@
+    });
+  });
+
+  describe('Appium config', () => {
+    describe('getAppiumConfig', () => {
+      it('should get a configuration object', async () => {
+        let config = await getAppiumConfig();
+        config.should.be.an('object');
+        should.exist(config['git-sha']);
+        should.exist(config.built);
+      });
+    });
+    describe('showConfig', () => {
+      let _info = logger.info;
+      let logOutput;
+      before(() => {
+        // need to be able to tell if logger.info was called
+        logger.info = function (...args) {
+          _info(args);
+          logOutput = args.join(' ');
+        };
+      });
+      after(() => {
+          logger.info = _info;
+        });
+        beforeEach(() => {
+          logOutput = undefined;
+        });
+      showConfig;
+      it('should log the config to logger.info', async () => {
+        let config = await getAppiumConfig();
+        await showConfig();
+        expect(logOutput).to.exist;
+        logOutput.should.contain(JSON.stringify(config));
+      });
+    });
+  });
+
+  describe('node.js config', () => {
+    let _process = process;
+    before(() => {
+      // need to be able to write to process.version
+      // but also to have access to process methods
+      // so copy them over to a writable object
+      let tempProcess = {};
+      for (let [prop, value] of _.pairs(process)) {
+        tempProcess[prop] = value;
+      }
+      process = tempProcess;
+    });
+    after(() => {
+      process = _process;
+    });
+    describe('checkNodeOk', () => {
+      it('should fail if node is below 0.10', () => {
+        process.version = 'v0.9.12';
+        checkNodeOk.should.throw;
+      });
+      it('should succeed if node is 0.10+', () => {
+        process.version = 'v0.10.0';
+        checkNodeOk.should.not.throw;
+      });
+    });
+
+    describe('warnNodeDeprecations', () => {
+      let called = false;
+      let _warn = logger.warn;
+      before(() => {
+        // need to be able to tell if logger.warn was called
+        logger.warn = function (...args) {
+          _warn(args);
+          called = true;
+        };
+      });
+      after(() => {
+        logger.warn = _warn;
+      });
+      beforeEach(() => {
+        called = false;
+      });
+      it('should log a warning if node is below 0.12', () => {
+        process.version = 'v0.9.12';
+        warnNodeDeprecations();
+        expect(called).to.be.true;
+      });
+      it('should not log a warning if node is 0.12+', () => {
+        process.version = 'v0.12.0';
+        warnNodeDeprecations();
+        expect(called).to.be.false;
+      });
+    });
+  });
+
+  describe('server arguments', () => {
+    let parser = getParser();
+    describe('getNonDefaultArgs', () => {
+      let args = {};
+      before(() => {
+        // give all the defaults
+        _.each(parser.rawArgs, (rawArg) => {args[rawArg[1].dest] = rawArg[1].defaultValue;});
```

Comment:
i've been moving away from `_.each` to actual `for...of` loops. i do like that this is on one line though.